src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -607,6 +596,20 @@ public List<ClusterTransactionCommand.SingleClusterDatabaseCommand> ExecuteClust
             return batch;
         }
 
+        private (List<ClusterTransactionCommand.SingleClusterDatabaseCommand> Batch, long MaxIndex) ReadCommandBatch(TransactionOperationContext context, int batchSize)
+        {
+            var batch = new List<ClusterTransactionCommand.SingleClusterDatabaseCommand>();
+            var maxIndex = -1L;
+            foreach (var command in ClusterTransactionCommand.ReadCommandsBatch(context, Name, fromCount: _nextClusterCommand, take: batchSize))
```

Comment:
where do we take into account the last cluster tx index here?

Reply:
Here - https://github.com/ravendb/ravendb/pull/18503/files#diff-76002270932dd2fbaba4ed54861fc2dc5c82ea677fea3ccdd23a4200a8f72fd5R608